modules/apps/commerce/commerce-product-content-search-web/src/main/resources/META-INF/resources/js/sort/view.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {createPortletURL, navigate} from 'frontend-js-web';
+
+function changeOrderBy(currentURL, portletDisplayId, label) {
+	const newUrl = createPortletURL(currentURL, {
+		orderByCol: label,
+		p_p_id: portletDisplayId,
+	});
+
+	navigate(newUrl.toString());
+}
+
+export default function ({currentURL, namespace, portletDisplayId}) {
+	const optionsLink = document.querySelectorAll(
+		`.dropdown-item.transition-link[id^=${namespace}]`
+	);
+
+	const commerceSortByButton = document.querySelector('.commerce-order-by');
+
+	const dropdownSortByButton = document.getElementById(
+		`${namespace}commerce-dropdown-order-by`
+	);
+
+	function handleOptionsLink(event, link) {
+		event.preventDefault();
+		changeOrderBy(
+			currentURL,
+			portletDisplayId,
+			link.getAttribute('data-label')
+		);
+	}
+
+	optionsLink.forEach((link) => {
+		link.addEventListener('click', handleOptionsLink(link));
+	});
+
+	window.addEventListener('mouseup', (event) => {
```

Comment:
In the clean-up function, we should get rid of all the listeners that might create problems once the portlet will disappear from the page.

Once `optionsLink` will be stripped from the page, the listener will disappear too.
Listeners applied to the window will remain, though.

Please give a name to this arrow function so you can remove it as a listener in the dispose  method